### DISCUSSION TITLE
[BTT Pad 7] Find help, and help others.

### DISCUSSION THREAD

**Discussion by bassamanator** [bassamanator/Sovol-SV06-firmware#181] [Announcements]
Please use this discussion to ask for help, and help others, with the BTT Pad.

Note that I'm not a BTT Pad user.

Links:
- Pad 7 [image](https://github.com/bigtreetech/Pad7/releases).

**Comment by tjhayko** [comment]
I'm not sure if this is the correct place to ask this, but I'm trying to get my SV06plus working with a BTT Pad 7 based on bassmanator's instructions, but /dev/serial doesn't seem to exist :(

I've flashed CB1_Debian11_Klipper_kernel5.16_20240319.img.xz onto my SD card, so I think I'm on the latest OS for the BTT Pad 7 with CB1

**Reply by tjhayko** [reply]
I've traced the problem with /dev/serial not existing back to bad USB cable (even though it worked with multiple other devices :( ), but now I get the following the klippy.log:

```
mcu 'mcu': Timeout on connect
mcu 'mcu': Wait for identify_response
Traceback (most recent call last):
  File "/home/biqu/klipper/klippy/serialhdl.py", line 68, in _get_identify_data
    params = self.send_with_response(msg, 'identify_response')
  File "/home/biqu/klipper/klippy/serialhdl.py", line 262, in send_with_response
    return src.get_response([cmd], self.default_cmd_queue)
  File "/home/biqu/klipper/klippy/serialhdl.py", line 319, in get_response
    self.serial.raw_send_wait_ack(cmds[-1], minclock, reqclock,
  File "/home/biqu/klipper/klippy/serialhdl.py", line 254, in raw_send_wait_ack
    self._error("Serial connection closed")
  File "/home/biqu/klipper/klippy/serialhdl.py", line 61, in _error
    raise error(self.warn_prefix + (msg % params))
serialhdl.error: mcu 'mcu': Serial connection closed
```

**Comment by tjhayko** [comment]
I think we can close this off.  In addition to a bad USB cable, I also had a bad flash of the printer firmware.  Using a different microSD card to flash the firmware seems to have fixed the issue, and I'm happily printing away (for the most part) now.

**Reply by bassamanator** [reply]
Turning a budget printer into a very reliable and consistent printer takes time and effort. Keep at it, because it's well worth it!

**Comment by Luciodrilo** [comment]
Muy buenas, a ver si alguien por aquí me puede ayudar, tengo una sovol sv 06+.
cansado de estar mete/saca tarjeta micro usb y previendo que se va a fastidiar la conexión de la impresora o de mi pc.
Pensé en hacer algo al respecto, directamente con un cable usb no funciona, no encontré ningún dispositivo con bluetoot o wifi para ponerle a la impresora, empecé a informarme y compre una BTT 7", ahora pensando que el aparato de marras seria conectarlo y mas o menos ya estaría (es lo que pensé por la falta de información que hay en internet y la totalmente inexistente por parte de la marca de la pantalla) resulta que hay que cambiar archivos de la impresora, (flasearla), mas sorpresas, sovol no los tiene, bigtreetech tampoco y ya con ganas de meterle fuego a todo y cabreadisimo (ojo, con nadie de por aquí solo por la situación en la que e terminado) e llegado a aqui, tendré suerte y me podréis ayudar??.
Pensad que ya tengo sesenta años y que aun que medio me defiendo con cosas de informatica acabo de descubrir linux (aun que no se si en realidad es un programa o solo sirve para acompañar la tortilla de patatas!!!).
Si alguien me pudiese indicar o pasarme algún link de un verdadero paso a paso (Para zopencos como yo!) de como flasear la impresora y de los ajustes que me puedan faltar les estaría eternamente agradecido.
Espero se puedan hacer estas consultas por aquí y si no es así perdón!!.

**Comment by Luciodrilo** [comment]
Y si no es mucho pedir, tengo entendido que el cable para conectar de laBTT a la impresora debe de ser solo de datos, pero no encuentro por ningún sitio, sabéis donde?.
O como modificarlo?, mil gracias.

**Reply by bassamanator** [reply]
You can find a `data` cable from Amazon, aliexpress. The following are listed as `data` cables (some are affiliate links):
- https://s.click.aliexpress.com/e/_oENGhGw
- https://www.aliexpress.com/item/1005003784002459.html

Flashing the motherboard is easy, you can find instructions [here](https://github.com/bassamanator/Sovol-SV06-firmware?tab=readme-ov-file#flash-firmware).

To clarify: you own a BTT Pad 7?

**Comment by Luciodrilo** [comment]
Hola, si .
Tengo una BTT pad 7".
Estuve viendo muchísimos comentarios sobre ella y sobre las bondades de poner klipper  en tu impresora, lo que no sabía es que el ictus que tuve me limitaría tanto y ahora resulta que : el inglés que sabía se me olvidó (aún puedo arreglarme con el traductor), se me olvidan las cosas tan rápido (tengo que anotarlo todo), cuando no consigo hacer lo que quiero me pongo nervioso y me sube mucho la tensión, debo dejarlo todo e ir corriendo a tomar la medicación.
El caso es que he llegado a poner en marcha la pad, pero al flasear la impresora (una sovol sv 06+) pues no funciona, algo he tenido que hacer mal, leyendo vi que tú habías echo el archivo .bin que es necesario para la misma impresora , sabrías decirme si valdría para todas las impresoras como la mía?.
Sabrías decirme en caso de que si sirva de donde descargarlo si lo tienes como público?.
Y muchas gracias por responderme y por tu tiempo.
Un saludo desde España.

Enviado desde Outlook para Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by bassamanator** [reply]
Sorry, missed this.

If you're asking whether the config for the sv06+ found on this repo will work with your Pad 7, the answer is yes. Unfortunately, I can't give you specific instructions on how to make this happen because I don't own a Pad 7. All you need is basic linux knowledge to make this happen.

**Comment by Luciodrilo** [comment]
Muchas gracias, acabé comprando una sv07+.
Si lo llego a saber antes, con lo que gaste en la pad7, en una Raspberry pi 3B+ con carcasa y pantalla de 7" táctil hubiera comprado la impresora, bueno, ahora la tengo guardada, pad 7 y Raspberry también.
Como pueden torcerse las cosas, yo solo quería poder mandar los archivos desde el pc a la impresora 3D, terminé acumulando trastos y comprando otra impresora!!.

Enviado desde Outlook para Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by lusciouslou91** [comment]
I have the Pad 7 and literally no knowledge of Linux. After about 5 hours, I am still no closer than the very first step. This guide is undecipherable to me. I am sure it is very well written, and wonderful for anyone with any amount of Linux experience. From absolute 0 this is impossible to understand. I thought that I could figure this out, but it does not seem so. Does anyone have the original files that were on the Pad 7? or give me the files that worked for you? Cause I don't understand any of this, and no matter what I try all I get is errors. Googling said errors results in no answers.

Thanks to anyone who can help.

**Reply by bassamanator** [reply]
The guide is written for people who have `zero` knowledge of linux, actually. It is a step by step guide for any linux distro, however, since the pad 7 is an _opinionated_ device, things might be a bit more complicated for a novice user. So, your frustration is understandable and expected. Hang in there!

You can find the official debian image (that's the operating system) [here](https://github.com/bigtreetech/Pad7/releases/tag/V1.0.0).

Here's a youtube video of [pad 7 + sovol printer](https://www.youtube.com/watch?v=AWaB4uA8aqk). [Here's](https://www.youtube.com/watch?v=pBk2D4ZiHMw) another that appears to go more in depth.

Keep at it!

**Reply by lusciouslou91** [reply]
You use a lot of terms that even after googling do not help explain what it means. You reference programs that I've downloaded and can't figure it how to use.
I'm pretty tech savvy, but your guide is indecipherable to me.

I've watched both of those vdeos multiple times and neither of them helped me.

**Reply by bassamanator** [reply]
I think it's best you seek help elsewhere, then.

Bye.
